### PR TITLE
More docs!

### PR DESCRIPTION
**Pull request by johnwcallahan**
## Description

This is the final PR of the docs spree I've been on for the last few weeks. New guides in this PR:

4. Component Library
7. Accessibility
8. Testing (from the existing TESTING.MD with minor edits)
9. Mocking Data (from the existing MOCKING_DATA.MD with an addition on API mocking)
10. Local Dev Tools (from DEV_TOOLS.MD with additions)
11. Feature Flags (new)
12. Managing Dependencies

I also took the error handling section of CODE_CONVENTIONS.MD and removed that file. 

This completes the first version of the development guide:
Screen Shot 2021-08-27 at 5 15 30 PM

### CODE REVIEW

File: docs/development-guide/04-component-library.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Component Library
+
+## Material-UI
+
+We use [Material-UI](https://material-ui.com/) as the primary component library for Cloud Manager. The library contains many UI primitives like `<Typography />` and `<Button />` as well as a layout system with the `<Grid />` component.
+
+All MUI components have abstractions in the manager codebase, meaning you will use relative imports to use them instead of importing from MUI directly:
```
Comment (tiffwong) [inline docs/development-guide/04-component-library.md:7]: Should `manager` be capitalized?

File: docs/development-guide/05-fetching-data.md [hunk 2]
```diff
@@ -157,3 +157,59 @@ const UsernameDisplay: React.FC<ProfileProps> = (props) => {
...
+
+#### API Error arrays
+
+Components making API requests generally expect to work with an array of Linode API errors. These have the shape:
+
+```js
+{ field: 'field-name', reason: 'why this error occurred' }
+```
+
+We have added an interceptor to our Axios instance that essentially guarantees that any error from an API function will have this shape. For example, if you block network requests using Chrome Dev Tools, there is no response from the API. But if you `.catch()` this error, you'll find that it has the above shape, with a default message ("An unexpected error occurred.")
```
Comment (tiffwong) [inline docs/development-guide/05-fetching-data.md:171]: Needs a period at the end of this sentence

File: docs/development-guide/09-mocking-data.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+This guide covers various methods of mocking data while developing or testing Cloud Manager.
+
+## Mocking APIv4
+
+Often when developing a feature you'll need your account or resources to be in a specific state. In other words, you'll need to be receiving specific data from the API.
+
+The best way to do this is to _mock the API_. This is made simple using **factories** and the **mock service worker**.
+
+**Factories**
+We use [factory.ts](https://www.npmjs.com/package/factory.ts) to generate mock data. With factory.ts you define a base "factory" for a given type, then use the factory to generate real TypeScript objects:
```
Comment (tiffwong) [inline docs/development-guide/09-mocking-data.md:12]: `factory.ts` should probably we surrounded by ticks

**Review by tiffwong — APPROVED** [review]
More greatness! 💯

**Review by ndukatz — APPROVED** [review]
Read through all of it and it looks great. All of your new documentation has very useful!

File: docs/development-guide/09-mocking-data.md (hunk 3 above)
Comment (johnwcallahan) [reply docs/development-guide/09-mocking-data.md:12]: I think this one is OK since "factory.ts" is the name of the library.

**Pull request merged**